packages/techdocs-common/src/stages/publish/local.ts

```diff
@@ -141,12 +141,19 @@ export class LocalPublish implements PublisherBase {
     return express.static(staticDocsDir);
   }
 
+  async getStorageUrl() {
+    return (
+      this.config.getOptionalString('techdocs.storageUrl') ??
+      (await this.discovery.getBaseUrl('techdocs'))
+    );
+  }
+
   async hasDocsBeenGenerated(entity: Entity): Promise<boolean> {
     const namespace = entity.metadata.namespace ?? 'default';
     return new Promise(resolve => {
-      this.discovery.getBaseUrl('techdocs').then(techdocsApiUrl => {
+      this.discovery.getBaseUrl('techdocs').then(async techdocsApiUrl => {
         const storageUrl = new URL(
-          new URL(this.config.getString('techdocs.storageUrl')).pathname,
+          new URL(await this.getStorageUrl()).pathname,
```

Comment:
Huh, I didn't notice this but the existing implementation was kind of unnecessary.

Local Publisher does not need to call an API to fetch the files. It is directly responsible for communicating with the local filesystem. So, it should instead use the `staticDocsDir` and read the file.